docs/source/tutorial.rst

```diff
@@ -724,3 +724,36 @@ sample points for both functions:
 
     print(f"Quadrature results: {integral1}, {integral2}")
 
+Multidimensional/Vectorized Integrands
+--------------------------------------
+
+If you wish to evaluate many different integrands over the same domain, it may be faster to pass in a vectorized formulation if possible.
+
+As an example, here is an example that could be improved:
+
+.. code:: ipython3
+
+    def parametrized_integrand(x, a, b):
+        return torch.sin((a + b) * x)
+
+    a_params = torch.arange(10)
+    b_params = torch.arange(2, 10)
+    integration_domain = torch.Tensor([[0, 1]])
+    simp = Simpson()
+    result = torch.stack([torch.Tensor([simp.integrate(lambda x: parametrized_integrand(x, a, b), dim=1, N=101, integration_domain=integration_domain) for a in a_params]) for b in b_params])
+
+
+Now let's see how to do this a bit more simply, and in a way that provides signficant speedup as the size of the integrand's `grid` grows:
```

Comment:
Actually running this gives 0 speedup because the example is super small and simple (both compute virtually instantly). Maybe we can either evaluate more integrands in parallel or make the integrand more complex?

Reply:
E.g. running like this

```py
def parametrized_integrand(x, a, b):
    return torch.sqrt(torch.cos(torch.sin((a + b) * x)))

a_params = torch.arange(40)
b_params = torch.arange(10, 20)
integration_domain = torch.Tensor([[0, 1]])
simp = tq.Simpson()
result = torch.stack([torch.Tensor([simp.integrate(lambda x: parametrized_integrand(x, a, b), dim=1, N=101, integration_domain=integration_domain) for a in a_params]) for b in b_params])
```

and

```py
grid = torch.stack([torch.Tensor([a + b for a in a_params]) for b in b_params])

def integrand(x):
    return torch.sqrt(torch.cos(torch.sin(torch.einsum("i,jk->ijk", x.flatten(), grid))))

result_vectorized = simp.integrate(integrand, dim=1, N=101, integration_domain=integration_domain)

torch.all(torch.isclose(result_vectorized, result)) # True!
```

it goes from 188ms to 5ms on my machine 🙃

Reply:
Is there a suggestion here or are you happy it worked haha?

Reply:
I would suggest to increase the params and complexity of the function with the code I posted so the effect is more noticeable?

Reply:
Ah ok, I see.  Will do.